src/V3Task.cpp

```diff
@@ -1645,6 +1688,28 @@ V3TaskConnects V3Task::taskConnects(AstNodeFTaskRef* nodep, AstNode* taskStmtsp)
                 if (funcp->classMethod() && funcp->lifetime().isStatic()) newvaluep = funcRefp;
             } else if (AstConst* const constp = VN_CAST(portp->valuep(), Const)) {
                 newvaluep = constp;
+            } else if (AstNodeExpr* const exprp = VN_CAST(portp->valuep(), NodeExpr)) {
+                auto clone = new AstArg{exprp->fileline(), portp->name(), exprp->cloneTree(true)};
+                nodep->addPinsp(clone);
+                AstNodeModule* modulep = nullptr;
+                AstNode* abovep = nodep;
+                while (!modulep && abovep->backp()) {
+                    abovep = abovep->backp();
+                    modulep = VN_CAST(abovep, NodeModule);
+                    if (modulep) break;
+                }
+                UASSERT(modulep, "AstNodeFTaskRef not under AstNodeModule");
+                {
+                    ArgumentsRelinkVisitor v{clone, modulep, tconnects, ++tmpFTaskArgVarCnt};
+                    if (v.replaced()) {
+                        if (tconnects[i].second) {  // Have a "nullptr" pin already defined for it
+                            VL_DO_CLEAR(tconnects[i].second->unlinkFrBack()->deleteTree(),
+                                        tconnects[i].second = nullptr);
+                        }
+                        tconnects[i].second = clone;
```

Comment:
Have you tested a situation when we have 2 optional arguments (not provided in a call) and both refer to argument, which was provided in the call and its value has a side effect?
I mean a situation like this:
```systemverilog
function void func (int a, int b = a, int c = a);
endfunction
function int print_and_get_1;
   $display("ABCD");
   return 1;
endfunction
func(print_and_get_1());
```
I think that the side effect will be doubled, because there will be two assignments with this expressions.